File: vue-project/src/models/Challenge.ts
export enum ChallengeType {
    DAILY = "daily",
    SPECIAL = "special",
    MILESTONE = "milestone",
}

export interface Challenge {
    id: string;
    title: string;
    description: string;
    coins: number;
    goal: number;
    challengeType: ChallengeType;
    isLimited?: boolean;
    expiresIn?: number; // in seconds
}

File: FastStudy/models/Challenge.py
from beanie import Document
from typing import Optional, Dict
from enum import Enum
from pydantic import BaseModel


class ChallengeType(str, Enum):
    DAILY = "daily"
    SPECIAL = "special"
    MILESTONE = "milestone"


class TierName(str, Enum):
    BRONZE = "bronze"
    SILVER = "silver"
    GOLD = "gold"
    PLATINUM = "platinum"


class Challenge(Document):
    title: str
    description: str
    coins: int
    goal: int
    challenge_type: ChallengeType
    is_limited: bool = False
    expires_in: Optional[int] = None  # in seconds

    class Settings:
        name = "challenges"

class ProgressUnit(str, Enum):
    HOURS = "hours"
    DAYS = "days"
    BLOCKS = "blocks"

class TierRequirement(BaseModel):
    value: int
    coins: int
